src/_P002_ADC.ino

```diff
@@ -283,20 +341,70 @@ float P002_applyCalibration(struct EventStruct *event, float float_value) {
   return float_value;
 }
 
-void P002_performRead(struct EventStruct *event, uint16_t& value) {
+void P002_performRead(struct EventStruct *event, int& value) {
   #ifdef ESP8266
   value = espeasy_analogRead(A0);
   #endif // if defined(ESP8266)
   #if defined(ESP32)
-  value = espeasy_analogRead(CONFIG_PIN1);
+  value = espeasy_analogRead(CONFIG_PIN1, P002_READ_AS_TOUCH != P002_NO_TOUCH);
   #endif // if defined(ESP32)
 }
 
-void P002_formatStatistics(const String& label, int16_t raw, float float_value) {
+void P002_formatStatistics(const String& label, int raw, float float_value) {
   addRowLabel(label);
   addHtml(String(raw));
   html_add_estimate_symbol();
   addHtml(String(float_value, 3));
 }
 
+#ifdef ESP32
+
+/**********************************************************************************
+* Touch pin configuration
+**********************************************************************************/
+void P002_setEventParams(int pin, uint16_t threshold) {
+  int adc, ch, t;
+
+  if (getADC_gpio_info(pin, adc, ch, t)) {
+    switch (t) {
+      case 0: touchAttachInterrupt(T0, P002_got_T0, threshold); break;
+      case 1: touchAttachInterrupt(T1, P002_got_T1, threshold); break;
+      case 2: touchAttachInterrupt(T2, P002_got_T2, threshold); break;
+      case 3: touchAttachInterrupt(T3, P002_got_T3, threshold); break;
+      case 4: touchAttachInterrupt(T4, P002_got_T4, threshold); break;
+      case 5: touchAttachInterrupt(T5, P002_got_T5, threshold); break;
+      case 6: touchAttachInterrupt(T6, P002_got_T6, threshold); break;
+      case 7: touchAttachInterrupt(T7, P002_got_T7, threshold); break;
+      case 8: touchAttachInterrupt(T8, P002_got_T8, threshold); break;
+      case 9: touchAttachInterrupt(T9, P002_got_T9, threshold); break;
+    }
+  }
+}
+
+void P002_got_T0() IRAM_ATTR;
+void P002_got_T1() IRAM_ATTR;
+void P002_got_T2() IRAM_ATTR;
+void P002_got_T3() IRAM_ATTR;
+void P002_got_T4() IRAM_ATTR;
+void P002_got_T5() IRAM_ATTR;
+void P002_got_T6() IRAM_ATTR;
+void P002_got_T7() IRAM_ATTR;
+void P002_got_T8() IRAM_ATTR;
+void P002_got_T9() IRAM_ATTR;
+
+void P002_got_T0() { p002_pinTouched |= (1 << 0); }
+void P002_got_T1() { p002_pinTouched |= (1 << 1); }
+void P002_got_T2() { p002_pinTouched |= (1 << 2); }
+void P002_got_T3() { p002_pinTouched |= (1 << 3); }
+void P002_got_T4() { p002_pinTouched |= (1 << 4); }
+void P002_got_T5() { p002_pinTouched |= (1 << 5); }
+void P002_got_T6() { p002_pinTouched |= (1 << 6); }
+void P002_got_T7() { p002_pinTouched |= (1 << 7); }
+void P002_got_T8() { p002_pinTouched |= (1 << 8); }
+void P002_got_T9() { p002_pinTouched |= (1 << 9); }
```

Comment:
you can use it with bitSet macro like: `bitSet(p002_pinTouched, 9)`

Reply:
Yep that's perhaps a more readable one.

Reply:
Just updated it and also removed the functions I once made which were a bit (pun intended) more complex.